packages/emotion-primitives/src/convertToRNStyles.js

```diff
@@ -0,0 +1,109 @@
+import { StyleSheet } from 'react-primitives'
+import transform from 'css-to-react-native'
+
+// Copied and edited from @emotion/css
+// Cannot use css from @emotion/css directly because of unnecessary checks, __emotion_styles, and registering interpolations in cache.
+function handleInterpolation(interpolation, couldBeSelectorInterpolation) {
+  if (interpolation == null) {
+    return ''
+  }
+
+  switch (typeof interpolation) {
+    case 'boolean':
+      return ''
+    case 'function':
+      return handleInterpolation.call(
+        this,
+        this === undefined ? interpolation() : interpolation(this.props),
+        couldBeSelectorInterpolation
+      )
+
+    default:
+      return interpolation
+  }
+}
+
+// Return evaluated css string
+function createStyles(strings, ...interpolations) {
+  let stringMode = true
+  let styles = ''
+
+  if (strings == null || strings.raw === undefined) {
+    stringMode = false
+    styles += handleInterpolation.call(this, strings, false)
+  } else {
+    styles += strings[0]
+  }
+
+  interpolations.forEach(function(interpolation, i) {
+    styles += handleInterpolation.call(
+      this,
+      interpolation,
+      styles.charCodeAt(styles.length - 1) === 46
+    )
+
+    if (stringMode === true && strings[i + 1] !== undefined) {
+      styles += strings[i + 1]
+    }
+  }, this)
+
+  return styles
+}
+
+function convertStyles(str) {
+  if (typeof str === 'string' && str.length === 0) return str
+
+  const styleObj = []
+
+  const parsedString = str.split(';')
+
+  parsedString.forEach(style => {
+    if (typeof style === 'string') {
```

Comment:
this if statement seems unnecessary, `style` has to be a string because it's outcome of `str.split(';')`